solutions/ipetek/turtlesim_mouse_tracker.cpp

```diff
@@ -0,0 +1,92 @@
+#include<ros/ros.h>
+#include<geometry_msgs/Twist.h>
+#include<geometry_msgs/Point.h>
+#include<turtlesim/Pose.h>
+#define _USE_MATH_DEFINES
+#include <cmath>
+#include <math.h>
+#include<stdio.h>
+
+
+int SCREEN_X = 1920; //EDIT
+int SCREEN_Y = 1080; //EDIT
+int LINEAR_SPEED = 5;
+int ANGULAR_SPEED = 10;
+
+class MouseTurtleController{
+    private:
+        ros::NodeHandle node;
+        ros::Subscriber mouse_pos;
+        ros::Subscriber turt_pos;
+        ros::Publisher turt_vel;
+        ros::Timer turt_pub_timer;
+
+        turtlesim::Pose last_turtle_pos;
+        geometry_msgs::Point last_mouse_pos;
+
+    public:
+        MouseTurtleController(){
+            mouse_pos = node.subscribe("/mouse_position", 1, &MouseTurtleController::mouse_pos_callback, this);
+            turt_pos = node.subscribe("/turtle1/pose", 1, &MouseTurtleController::turtle_pos_callback, this);
+            turt_vel = node.advertise<geometry_msgs::Twist>("/turtle1/cmd_vel", 1);
+            turt_pub_timer = node.createTimer(ros::Duration(0.01), &MouseTurtleController::turt_pub_callback, this);
+        }
+    
+    private:
+        void mouse_pos_callback(const geometry_msgs::Point::ConstPtr& msg){
+            last_mouse_pos = *msg;
+        }
+        void turtle_pos_callback(const turtlesim::Pose::ConstPtr& msg){
+            last_turtle_pos = *msg;
+        }
+        void turt_pub_callback(const ros::TimerEvent& event){
+            geometry_msgs::Twist pub_data;
+            geometry_msgs::Point desk_to_turt = desktop_to_turtlesim(last_mouse_pos);
+            double curr_distance = get_distance(desk_to_turt);
+            double curr_ang = get_ang_distance(desk_to_turt);
+            if(curr_distance > 1){
+                pub_data.linear.x = LINEAR_SPEED;
```

Comment:
This controller will always run at the same speed towards the mouse. If you want smoother control, you can set this linear speed proportional to the distance.